pubsub/cloud-client/src/main/java/com/example/pubsub/SubscriberExample.java

```diff
@@ -28,18 +28,29 @@
 import java.util.concurrent.LinkedBlockingDeque;
 
 public class SubscriberExample {
+  static class MessageAndConsumer {
+    PubsubMessage message;
+    AckReplyConsumer consumer;
+
+    MessageAndConsumer(PubsubMessage message, AckReplyConsumer consumer) {
+      this.message = message;
+      this.consumer = consumer;
+    }
+  }
 
   // use the default project id
   private static final String PROJECT_ID = ServiceOptions.getDefaultProjectId();
 
-  private static final BlockingQueue<PubsubMessage> messages = new LinkedBlockingDeque<>();
+  private static final BlockingQueue<MessageAndConsumer> messages = new LinkedBlockingDeque<>();
+
 
   static class MessageReceiverExample implements MessageReceiver {
 
     @Override
     public void receiveMessage(PubsubMessage message, AckReplyConsumer consumer) {
-      messages.offer(message);
-      consumer.ack();
+      if (!messages.offer(new MessageAndConsumer(message, consumer))) {
```

Comment:
Can you add a comment to the user explaining the logic? (e.g., "Attempt to insert the message into the queue. If unable, signal that it has failed to process."  

Also, should we consider a timeout here? The cost to receive/`nack()` isn't free, which means this could cause some thrashing if any customers copy this example verbatim.